### PR TITLE
Fix broken RSMTool CI build

### PR DESCRIPTION
**Pull request by Frost45**
Made changes to suppress all warnings when not in STRICT mode instead of just deprecation warnings.

### CODE REVIEW

File: rsmtool/test_utils.py [hunk 1]
```diff
@@ -790,20 +790,26 @@ def check_report(html_file,
     report_errors = 0
     report_warnings = 0
 
+    # Setting raise_warnings to false if not in STRICT mode
+    if IGNORE_WARNINGS:
+        raise_warnings = False
+
     with open(html_file, 'r') as htmlf:
-        for line in htmlf:
+        for i, line in enumerate(htmlf):
```
Comment (desilinguist) [inline rsmtool/test_utils.py:798]: What is the enumerate for? I don't see `i` being used anywhere?

**Review by desilinguist — APPROVED** [review]
This looks good @Frost45! One minor comment and a question: you had the same errors in `main` as `stable` before this fix and they were all resolved once this fix was applied?

File: rsmtool/test_utils.py [hunk 2]
```diff
@@ -141,10 +141,10 @@ def check_run_experiment(source,
     check_report(html_report, raise_warnings=False)
 
     # make sure that there are no warnings in the report
-    # but ignore deprecation warnings if appropriate
+    # but ignore warnings if appropriate
     warning_msgs = collect_warning_messages_from_report(html_report)
-    if IGNORE_DEPRECATION_WARNINGS:
-        warning_msgs = [msg for msg in warning_msgs if 'DeprecationWarning' not in msg]
+    if IGNORE_WARNINGS:
+        warning_msgs = [msg for msg in warning_msgs if 'Warning' not in msg]
```
Comment (mulhod) [inline rsmtool/test_utils.py:147]: Just wondering, won't "Warning" always be in the message? Which messages would not be filtered out here?

File: rsmtool/test_utils.py (hunk 1 above)
Comment (Frost45) [reply rsmtool/test_utils.py:798]: @desilinguist Yeah, there were 70 errors or so, when not in STRICT mode and 90 in STRICT mode (With deprecation warnings). After this fix, running the tests normally does not throw any errors.

File: rsmtool/test_utils.py (hunk 2 above)
Comment (mulhod) [reply rsmtool/test_utils.py:147]: Same for the repetitions below.

File: rsmtool/test_utils.py (hunk 1 above)
Comment (Frost45) [reply rsmtool/test_utils.py:798]: Oh, I was trying out something but ended up not needing it. Should I revert this line?

Comment (desilinguist) [reply rsmtool/test_utils.py:798]: Yes, please go ahead.

**Review by mulhod — APPROVED** [review]
Looks good! Just a question.

File: rsmtool/test_utils.py (hunk 2 above)
Comment (desilinguist) [reply rsmtool/test_utils.py:147]: Good question! I don't think we need this filtering since previously it was to filter out non-deprecation warnings :)

**Review by aloukina — CHANGES_REQUESTED** [review]
We also need to update documentation here: https://rsmtool.readthedocs.io/en/stable/internal/release_process.html?highlight=warnings#rsmtool-release-process to remove "deprecation"

File: rsmtool/test_utils.py (hunk 2 above)
Comment (Frost45) [reply rsmtool/test_utils.py:147]: @desilinguist @mulhod I was actually considering not calling `collect_warning_messages_from_report()` when not in STRICT mode but did not want to change it too much. There was some issue with blank spaces, but I've fixed that. What do you guys think?

Comment (Frost45) [reply rsmtool/test_utils.py:147]:
```suggestion
            if not IGNORE_WARNINGS:
                  warning_msgs = collect_warning_messages_from_report(html_report)
                  assert_equal(len(warning_msgs), 0)
```

Comment (mulhod) [reply rsmtool/test_utils.py:147]: @Frost45 Looks good to me!

Comment (mulhod) [reply rsmtool/test_utils.py:147]: Please make the change and let the build run.

File: rsmtool/test_utils.py [hunk 3]
```diff
@@ -231,10 +231,10 @@ def check_run_evaluation(source,
     check_report(html_report, raise_warnings=False)
 
     # make sure that there are no warnings in the report
-    # but ignore deprecation warnings if appropriate
+    # but ignore warnings if appropriate
     warning_msgs = collect_warning_messages_from_report(html_report)
-    if IGNORE_DEPRECATION_WARNINGS:
```
[the remark this answers is not in the document]
Comment (Frost45) [reply rsmtool/test_utils.py:236]: @aloukina I've made some changes to this code. Do you still think it should be made into a function?

**Review by desilinguist — APPROVED** [review]
Latest changes look great!!! 🎉

**Review by mulhod — APPROVED** [review]
LGTM!

File: rsmtool/test_utils.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (aloukina) [reply rsmtool/test_utils.py:236]: There is some repetition in the code, but I think it's fine to leave it as is.

**Review by aloukina — APPROVED** [review]
Thank you for doing this, looks great!

**Pull request merged**